controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java

```diff
@@ -36,33 +36,39 @@
         StatusUpdate execute(Integration model);
 
         class StatusUpdate {
-            private Optional<Integration.Status> status;
-            private Optional<String> statusMessage;
+            private Integration.Status status;
+            private String statusMessage;
+            private List<String> stepsPerformed;
 
-            public StatusUpdate(Optional<Integration.Status> status, Optional<String> statusMessage) {
+            public StatusUpdate(Integration.Status status, String statusMessage, List<String> stepsPerformed) {
                 this.status = status;
                 this.statusMessage = statusMessage;
-            }
-
-            public StatusUpdate(Optional<Integration.Status> status, String statusMessage) {
-                this(status, Optional.of(statusMessage));
+                this.stepsPerformed = stepsPerformed;
             }
 
             public StatusUpdate(Integration.Status status, String statusMessage) {
-                this(Optional.of(status), Optional.of(statusMessage));
+                this(status, statusMessage, null);
             }
 
             public StatusUpdate(Integration.Status status) {
-                this(Optional.of(status), Optional.empty());
+                this(status, null, null);
+            }
+
+            public StatusUpdate(Integration.Status status, List<String> stepsPerformed) {
+                this(status, null, stepsPerformed != null && stepsPerformed.size() > 0 ? stepsPerformed : null);
             }
 
-            public Optional<Integration.Status> getStatus() {
+            public Integration.Status getStatus() {
                 return status;
             }
 
-            public Optional<String> getStatusMessage() {
+            public String getStatusMessage() {
```

Comment:
My ide warns me to use Optional<> as fields or even method parameters. And there are people backing this opinion:

* http://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments
* http://dolszewski.com/java/java-8-optional-use-cases/
... and so on

personally, I think we should not overuse it, so it's hopefully ok to fix this where it is used. Its a helper multi-valued return value anyway.